x-pack/plugins/infra/public/containers/map/with_map.ts

```diff
@@ -52,8 +55,15 @@ export const withMap = graphql<
       },
       filters: [],
       metrics: [{ type: InfraMetricType.count }],
-      path: [{ id: 'n1', type: InfraPathType.hosts }],
+      path: [
+        { id: uuid.v1(), type: InfraPathType.terms, field: 'metricset.module' },
+        { id: uuid.v1(), type: InfraPathType.terms, field: 'metricset.name' },
+        { id: uuid.v1(), type: InfraPathType.hosts },
+      ],
     },
   }),
-  props: ({ data }) => ({ map: (data && data.source && data.source.map) || {} }),
+  props: ({ data }) => {
+    const nodes = get(data, 'source.map.nodes', []);
+    return { map: nodesToWaffleMap(nodes) };
```

Comment:
According to the schema `source.map.nodes` can be `null`, which means the `nodesToWaffleMap` function fails at runtime when the query does not yield any nodes:

![image](https://user-images.githubusercontent.com/973741/43953184-796d51ee-9c98-11e8-9fa9-ef00866f04e6.png)

The compiler would have warned about this if instead of the hand-written `Response`, the generated `Query.Query` (or `MapQuery.Query` after the proposed renaming above) from `common/graphql/types.ts` had been used.

How about fixing that by

1. use the generated query response type and
2. change the schema and resolver of `InfraResponse` to always return an array for `nodes` (empty when no nodes are found)?

The same goes for the `Variables`. Let's use the generated `Query.Variables` (or `MapQuery.Variables`).